dags/airflow-log-cleanup.py

```diff
@@ -0,0 +1,253 @@
+SCHEDULE_INTERVAL = "@daily"
+# Who is listed as the owner of this DAG in the Airflow Web Server
+DAG_OWNER_NAME = "operations"
+# List of email address to send email alerts to if this job fails
+ALERT_EMAIL_ADDRESSES = ["davidtnfsh@gmail.com"]
+# Length to retain the log files if not already provided in the conf. If this
+# is set to 30, the job will remove those files that are 30 days old or older
+DEFAULT_MAX_LOG_AGE_IN_DAYS = Variable.get(
+    "airflow_log_cleanup__max_log_age_in_days", 30
+)
+# Whether the job should delete the logs or not. Included if you want to
+# temporarily avoid deleting the logs
+ENABLE_DELETE = True
+# The number of worker nodes you have in Airflow. Will attempt to run this
+# process for however many workers there are so that each worker gets its
+# logs cleared.
+NUMBER_OF_WORKERS = 1
+DIRECTORIES_TO_DELETE = [BASE_LOG_FOLDER]
+ENABLE_DELETE_CHILD_LOG = Variable.get(
+    "airflow_log_cleanup__enable_delete_child_log", "False"
+)
+LOG_CLEANUP_PROCESS_LOCK_FILE = "/tmp/airflow_log_cleanup_worker.lock"
+logging.info("ENABLE_DELETE_CHILD_LOG  " + ENABLE_DELETE_CHILD_LOG)
+
+if not BASE_LOG_FOLDER or BASE_LOG_FOLDER.strip() == "":
+    raise ValueError(
+        "BASE_LOG_FOLDER variable is empty in airflow.cfg. It can be found "
+        "under the [core] (<2.0.0) section or [logging] (>=2.0.0) in the cfg file. "
+        "Kindly provide an appropriate directory path."
+    )
+
+if ENABLE_DELETE_CHILD_LOG.lower() == "true":
+    try:
+        CHILD_PROCESS_LOG_DIRECTORY = conf.get(
+            "scheduler", "CHILD_PROCESS_LOG_DIRECTORY"
+        )
+        if CHILD_PROCESS_LOG_DIRECTORY != " ":
+            DIRECTORIES_TO_DELETE.append(CHILD_PROCESS_LOG_DIRECTORY)
+    except Exception as e:
+        logging.exception(
+            "Could not obtain CHILD_PROCESS_LOG_DIRECTORY from "
+            + "Airflow Configurations: "
+            + str(e)
+        )
+
+default_args = {
+    "owner": DAG_OWNER_NAME,
+    "depends_on_past": False,
+    "email": ALERT_EMAIL_ADDRESSES,
+    "email_on_failure": True,
+    "email_on_retry": False,
+    "start_date": START_DATE,
+    "retries": 1,
+    "retry_delay": timedelta(minutes=1),
+}
+
+dag = DAG(
+    DAG_ID,
+    default_args=default_args,
+    schedule_interval=SCHEDULE_INTERVAL,
+    start_date=START_DATE,
+    tags=["teamclairvoyant", "airflow-maintenance-dags"],
+    template_undefined=jinja2.Undefined,
+)
+if hasattr(dag, "doc_md"):
+    dag.doc_md = __doc__
+if hasattr(dag, "catchup"):
+    dag.catchup = False
+
+start = DummyOperator(task_id="start", dag=dag)
+
+log_cleanup = (
+    """
+
+echo "Getting Configurations..."
+BASE_LOG_FOLDER="{{params.directory}}"
+WORKER_SLEEP_TIME="{{params.sleep_time}}"
+
+sleep ${WORKER_SLEEP_TIME}s
+
+MAX_LOG_AGE_IN_DAYS="{{dag_run.conf.maxLogAgeInDays}}"
+if [ "${MAX_LOG_AGE_IN_DAYS}" == "" ]; then
+    echo "maxLogAgeInDays conf variable isn't included. Using Default '"""
+    + str(DEFAULT_MAX_LOG_AGE_IN_DAYS)
```

Comment:
I must admit it becomes a bit difficult to read when it comes to lots of string concatenation.  Is it possible to use f-string or jinja to achieve the same result?